src/main/scala/swiss/dasch/FetchAssetPermissions.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2021 - 2024 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package swiss.dasch
+
+import cats.implicits._
+import sttp.capabilities.zio.ZioStreams
+import sttp.client3.SttpBackend
+import sttp.client3.*
+import sttp.client3.httpclient.zio.HttpClientZioBackend
+import swiss.dasch.config.Configuration
+import swiss.dasch.domain.AssetInfo
+import zio.*
+import zio.json.DecoderOps
+import zio.json.DeriveJsonDecoder
+import zio.json.JsonDecoder
+
+import scala.concurrent.duration._
+
+import FetchAssetPermissions.PermissionResponse
+
+trait FetchAssetPermissions {
+  def getPermissionCode(
+    jwt: String,
+    assetInfo: AssetInfo,
+  ): Task[Int]
+}
+
+class FetchAssetPermissionsLive(
+  sttp: SttpBackend[Task, ZioStreams],
+  apiConfig: Configuration.ApiConfig,
+) extends FetchAssetPermissions {
+  def getPermissionCode(
+    jwt: String,
+    assetInfo: AssetInfo,
+  ): Task[Int] =
+    (for {
+      uri <-
+        ZIO.succeed(
+          uri"http://${apiConfig.host}:${apiConfig.port}/admin/files/${assetInfo.assetRef.belongsToProject}/${assetInfo.derivative.filename}",
+        )
+      response    <- sttp.send(basicRequest.get(uri).header("Authorization", s"Bearer ${jwt}"))
+      successBody <- ZIO.fromEither(response.body).mapError(httpError(uri.toString, response.code.code, _))
+      permissionCode <-
+        ZIO.fromEither(successBody.fromJson[PermissionResponse].bimap(e => throw new Exception(e), _.permissionCode))
```

Comment:
I'm not so familiar with the sttp client, and what the bimap function does here. But the `throw` looks suspicious here - if this actually throws the exception, then that will result in a `ZIO.die` not in a `ZIO.fail`, like the return type `Task[Int]` would suggest.  
Have you tested this error case and verifies that it behaves as expected?

Reply:
Yeah, throw is not necessary here.